packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -314,6 +315,15 @@ class HotRunner extends ResidentRunner {
     bool enableDevTools = false,
     String route,
   }) async {
+    File mainFile = globals.fs.file(mainPath);
```

Comment:
Similar comment to the devfs comment. If you lower the lookup into the resident compiler that would reduce the number of places this logic needs to be repeated.

Reply:
Done